processing/src/main/java/io/druid/segment/QueryableIndexStorageAdapter.java

```diff
@@ -341,7 +435,18 @@ public void advance()
                       if (Thread.interrupted()) {
                         throw new QueryInterruptedException(new InterruptedException());
                       }
-                      cursorOffset.increment();
+                      cursorOffsetHolder.get().increment();
+
+                      while (!isDone()) {
```

Comment:
based on the benchmarks I ran (using unfiltered queries, just reading rows from the cursor), adding this indirection didn't have much of an impact on QueryableIndex read performance